File: Gerontocracy.Web/gerontocracy/modules/affair/models/vorfall-detail.ts
import { ReputationType } from '../../shared/models/reputation-type';
import { QuelleOverview } from './quelle-overview';
import { PolitikerOverview } from './politiker-overview';
import { VoteType } from './vote-type';

export interface VorfallDetail {
    id: number;
    titel: string;
    beschreibung: string;
    erstelltAm: Date;
    reputationType: ReputationType;
    reputation: number;
    quellen: QuelleOverview[];
    politikerOverview: PolitikerOverview;
    userVote: VoteType;
}

File: Gerontocracy.App/Models/Affair/VorfallDetail.cs
using System;
using System.Collections.Generic;

using Gerontocracy.App.Models.Party;
using Gerontocracy.App.Models.Shared;

namespace Gerontocracy.App.Models.Affair
{
    /// <summary>
    /// represents an affair detail dto
    /// </summary>
    public class VorfallDetail
    {
        /// <summary>
        /// Identifier
        /// </summary>
        public long Id { get; set; }

        /// <summary>
        /// Represents the title of the affair
        /// </summary>
        public string Titel { get; set; }

        /// <summary>
        /// The affair description
        /// </summary>
        public string Beschreibung { get; set; }

        /// <summary>
        /// When was the affair-entry created
        /// </summary>
        public DateTime ErstelltAm { get; set; }

        /// <summary>
        /// Is this good or bad?
        /// </summary>
        public ReputationType ReputationType { get; set; }

        /// <summary>
        /// How did the User vote?
        /// </summary>
        public VoteType? UserVote { get; set; }

        /// <summary>
        /// Score
        /// </summary>
        public int Reputation { get; set; }

        /// <summary>
        /// Sources
        /// </summary>
        public List<QuelleOverview> Quellen { get; set; }

        /// <summary>
        /// Politician involved
        /// </summary>
        public PolitikerOverview Politiker { get; set; }
    }
}
